src/plugins/simulator/visualizations/qt-opengl/qtopengl_camera.cpp

```diff
@@ -67,308 +64,207 @@ namespace argos {
-      }
-   }
+   CQTOpenGLCamera::~CQTOpenGLCamera() {}
    
    /****************************************/
    /****************************************/
 
    void CQTOpenGLCamera::Init(TConfigurationNode& t_tree) {
-      if(NodeExists(t_tree, "camera")) {
-         try {
-            TConfigurationNode tCameraNode;
-            tCameraNode = GetNode(t_tree, "camera");
-            /* Parse "placement" nodes */
+      try {
+         if(NodeExists(t_tree, "placements")) {
+            TConfigurationNode& tPlacementsNode = GetNode(t_tree, "placements");
+            /* Parse the placement nodes */
             TConfigurationNodeIterator itPlacement("placement");
-            SInt32 nIdx;
-            for(itPlacement = itPlacement.begin(&tCameraNode);
-                itPlacement != itPlacement.end();
-                ++itPlacement) {
-               GetNodeAttribute(*itPlacement, "idx", nIdx);
-               if(nIdx >=0 && nIdx <= 11) {
-                  m_sPlacement[nIdx].Init(*itPlacement);
+            for(itPlacement = itPlacement.begin(&tPlacementsNode);
+                  itPlacement != itPlacement.end();
+                  ++itPlacement) {
+               UInt32 unIndex = 0;
+               GetNodeAttribute(*itPlacement, "index", unIndex);
+               if(unIndex < m_arrPlacements.size()) {
+                  m_arrPlacements[unIndex].Init(*itPlacement);
                }
                else {
-                  THROW_ARGOSEXCEPTION("Error initializing QTOpenGL camera placement: value given for 'idx' is out of bounds. Value = \"" << nIdx << "\", allowed [0-11].");
+                  THROW_ARGOSEXCEPTION("Placement index is out of bounds");
                }
             }
-            /* Parse "timeline" nodes */
-            TConfigurationNodeIterator itTimeline("timeline");
-            for(itTimeline = itTimeline.begin(&tCameraNode);
-                itTimeline != itTimeline.end();
-                ++itTimeline) {
+         }
+         else {
+            const CVector3& cArenaSize = 
+               CSimulator::GetInstance().GetSpace().GetArenaSize();
+            const CVector3& cArenaCenter = 
+               CSimulator::GetInstance().GetSpace().GetArenaCenter(); 
+            for(UInt32 un_index = 0; 
+                un_index < m_arrPlacements.size();
+                ++un_index) {
+               CRadians cAngle(CRadians::PI_OVER_SIX);
+               cAngle *= static_cast<Real>(un_index);
+               m_arrPlacements[un_index].Position = 
+                  CVector3::X * cArenaSize.Length();
+               m_arrPlacements[un_index].Position.RotateZ(cAngle);
+               m_arrPlacements[un_index].Position += cArenaCenter;
+               m_arrPlacements[un_index].Target = cArenaCenter;
+               m_arrPlacements[un_index].Target.SetZ(0);
+               m_arrPlacements[un_index].Up = 
+                  /* position - target? */
+                  CVector3(0.0, 0.0, 1.0).Normalize();
```

Comment:
Use CVector3::Z

Reply:
Fixed in 1588d020fc01cde3cc3ef7750bbbb8b200b85b44